hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapFetchIndexOperation.java

```diff
@@ -130,20 +122,43 @@ protected void runInternal() {
         }
     }
 
+    @Nonnull
+    public static InternalIndex getInternalIndex(@Nonnull MapContainer mapContainer,
+                                                 @Nonnull String mapName,
+                                                 @Nonnull String indexName) {
+        Indexes indexes = mapContainer.getIndexes();
+        if (indexes == null) {
```

Comment:
You can use `mapContainer.isGlobalIndexEnabled()`

Reply:
https://github.com/hazelcast/hazelcast/pull/25328/commits/45addd1c7ddb68a7df779f4ea84167154469b7c4, added also in some other places to more easily find spots where we need to add local index support